### DISCUSSION TITLE
Feature request: add OpenTofu

### DISCUSSION THREAD

**Discussion by maazghani** [devcontainers/community#72] [Ideas]
I’ve created an [issue](https://github.com/devcontainers/features/issues/695) in the [devcontainer/features](https://github.com/devcontainers/features) repository for adding support for [OpenTofu](https://github.com/opentofu/opentofu), the open source alternative to Terraform and created a Draft PR in anticipation of the first release of OpenTofu.

### Questions:

* What does Terraform’s change from MPL to BUSL license mean for cases like the devcontainers/features which uses an MIT license that allows commercial-use, while BUSL does not?

* Should the PR go as far as removing the terraform feature? 

* Have we ever removed a feature? How would it work?

**Comment by samruddhikhandale** [comment]
Hi 👋 

Thanks for starting this discussion, really appreciate it. 

> What does Terraform’s change from MPL to BUSL license mean for cases like the devcontainers/features which uses an MIT license that allows commercial-use, while BUSL does not?

> Should the PR go as far as removing the terraform feature?

I am currently discussing it with the devcontainers team on what approach should be taken for the Terraform Feature, would appreciate your patience. Thank you!

> Have we ever removed a feature? How would it work?

We have added support for marking a Feature as deprecated, see `deprecated` property in https://containers.dev/implementors/features/. However, I don't think we have used this before. We are still not sure if we want to deprecate it, we'll post more details soon. 

deprecated | boolean | Indicates that the Feature is deprecated, and will not receive any further updates/support. This property is intended to be used by the supporting tools for highlighting Feature deprecation.
-- | -- | --

**Comment by samruddhikhandale** [comment]
Updates: The dev containers spec or the Features doesn't says that a dev container Feature(s) has to only install OSS components. The MIT license is for the Feature which is just an installer, and we install from an official source. What a user installs is governed by its own license and we are not redistributing by persisting it ourself. Hence, there are no changes/deprecations to be made to the Terraform Feature as it's still a legit option, thus, we have https://github.com/devcontainers/features/pull/697 opened to note the licensing changes in the README file.

> Should the PR go as far as removing the terraform feature?

We shouldn't be deprecating the Terraform Feature, and for adding support for the `OpenTofu` as it's new its a bit unclear if the community will embrace it or not. Hence, we can keep https://github.com/devcontainers/features/issues/695 open and reiterate on our decision based on community interest/feedbacks.

Let us know if you any followup questions/concerns. Thank you!

**Comment by maazghani** [comment]
Thank you for the comprehensive response. I appreciate the time taken to discuss the implications of the licensing as I’m sure this won’t be the last time it comes up.

Thanks again and I look forward to further updates and discussions on this topic.
